compiler/quilt/tools/main.py

```diff
@@ -152,8 +147,7 @@ def check_hash(group, hashstr):
     install_p.set_defaults(func=command.install)
     install_p.add_argument("-f", "--force", action="store_true", help="Overwrite without prompting")
     install_group = install_p.add_mutually_exclusive_group()
-    install_group.add_argument("-x", "--hash", help="Package hash",
-                               type=lambda val: check_hash(install_p, val))
```

Comment:
Does this change anything? I'm guessing the existing code fails sooner, during argument parsing.

Reply:
Yup, removing that and moving its effects into _match_hash have a few nice effects.  The existing code fails sooner, but it's not necessarily preferable. 

The notable effects of the change are:
* more succinct code with no loss of clarity (argparse 'type' isn't hijacked for cleanup)
* removal of code duplication (otherwise, hash needs to be checked in `main` and `command`)
* provides specific choke point for handling hash scrubbing (short->long, lower(), hash lookup, ambiguous hash errors, invalid/missing hash errors)
* clearer error messages (syntax errors belong in argparse, imo, but for a correctly-placed but invalid token, showing argparse help again just seems distracting)

Reply:
Ah, ok then